torch/_jit_internal.py

```diff
@@ -655,21 +659,18 @@ class Future(Generic[T]):
     def __init__(self, types):
         self.__args__ = types
 
-class RRef(Generic[T]):
-    __slots__ = ['__args__']
-
-    def __init__(self, types):
-        self.__args__ = types
-
 def is_future(ann):
     if ann is Future:
         raise RuntimeError('Attempted to use torch.jit.Future without a '
                            'contained type. Please add a contained type, e.g. '
                            'torch.jit.Future[int]')
     return getattr(ann, "__origin__", None) is Future
 
-def is_rref(ann):
-    return getattr(ann, "__origin__", None) is RRef
+if torch.distributed.rpc.is_available():
```

Comment:
If this file is supposed to be used without RPC. Does the following work?

```python
if torch.distributed.is_available() and torch.distributed.rpc.is_available():
    from torch.distributed.rpc import RRef

    def is_rref(ann):
        return getattr(ann, "__origin__", None) is RRef
```

If this file should always run in conjunction with RPC, then we can error out when RPC is not available?

Reply:
This file does not necessary requires RPC. We should hie RRef type if RPC module is not available. This file used to work, because typing system is at the bottom of the dependency stack. so even RPC module is not loaded, we can define/register/load a RRef type, but this RRef type does not work for real purpose.

Reply:
also, why you think `torch.distributed.is_available()` is necessary? also they are controlled by a single build flag. But I think RPC module could be used without `torch.distributed` C++ bindings being initialized, right?

Reply:
I was referring to dropping `import torch.distributed.rpc` on top of this file, and then do the check + guarded import here.

Reply:
> Why you think torch.distributed.is_available() is necessary?

I was puzzled about the comment below:

> # This is needed. Python "import" statement executes "torch/distributed/__init__.py",
> # which installs C++ bindings by calling `torch._C._c10d_init()`.
> # Otherwise, "AttributeError: module 'torch' has no attribute 'distributed'" is raised.

Why not loading c++ binding could make `distributed` package disappear?